src/var_equality.jl

```diff
@@ -57,16 +57,19 @@ function Base.show(io::IOContext, t::VarianceEqualityTest)
     end
 end
 
-function anova(scores::AbstractVector{<:Real}...)
-    Nᵢ = [length(g) for g in scores]
+function anova(scores::AbstractVector{<:AbstractVector})
```

Comment:
It seems a (minor but still) regression to allocate a vector in the multi-arg version. Maybe we could just allow iterators here, something like
```suggestion
function anova(scores)
    if isempty(scores) || !all(x -> x isa AbstractVector{<:Real}, scores)
        throw(ArgumentError("`anova` requires a non-empty collection of vectors of `Real` numbers"))
    end
```

Reply:
Thank you for the review.
I dropped the `scores`/`groups` type constraints, as you suggested, and tried to avoid vector allocations, where possible.
Considering the other code tweaks, this PR should reduce the allocations when constructing `VarianceEqualityTest` objects.